be/src/vec/functions/function_compress_varchar.cpp

```diff
@@ -0,0 +1,130 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstddef>
+#include <limits>
+#include <type_traits>
+
+#include "common/exception.h"
+#include "common/status.h"
+#include "runtime/primitive_type.h"
+#include "vec/columns/column_const.h"
+#include "vec/columns/column_vector.h"
+#include "vec/core/types.h"
+#include "vec/data_types/data_type.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/data_types/data_type_string.h"
+#include "vec/functions/function.h"
+#include "vec/functions/function_helpers.h"
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris::vectorized {
+
+struct CompressAsSmallInt {
+    static constexpr auto name = "compress_as_smallint";
+};
+
+struct CompressAsInt {
+    static constexpr auto name = "compress_as_int";
+};
+
+struct CompressAsBigInt {
+    static constexpr auto name = "compress_as_bigint";
+};
+
+struct CompressAsLargeInt {
+    static constexpr auto name = "compress_as_largeint";
+};
+
+template <typename Name, typename ReturnType>
+class FunctionCompressVarchar : public IFunction {
+private:
+    static inline void reverse_bytes(uint8_t* __restrict s, size_t length) {
+        if (length == 0) {
+            return;
+        }
+
+        int c, i, j;
+
+        for (i = 0, j = length - 1; i < j; i++, j--) {
+            c = s[i];
+            s[i] = s[j];
+            s[j] = c;
+        }
```

Comment:
Would it better to use `std::reverse(s, s + length);` instead?

Reply:
can `std::reverse` be applied to raw pointer?